src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs

```diff
@@ -421,34 +411,30 @@ public async Task StopAsync(CancellationToken cancellationToken = default)
 
         public async ValueTask DisposeAsync()
         {
-            // The user didn't change the ContentRootFileProvider instance, we can dispose it
-            if (ReferenceEquals(_hostEnvironment.ContentRootFileProvider, _defaultProvider))
-            {
-                // Dispose the content provider
-                await DisposeAsync(_hostEnvironment.ContentRootFileProvider).ConfigureAwait(false);
-            }
-            else
+            IFileProvider contentRootFileProvider = _hostEnvironment.ContentRootFileProvider;
+            await DisposeAsync(contentRootFileProvider).ConfigureAwait(false);
+
+            if (!ReferenceEquals(contentRootFileProvider, _defaultProvider))
             {
                 // In the rare case that the user replaced the ContentRootFileProvider, dispose it and the one
                 // we originally created
-                await DisposeAsync(_hostEnvironment.ContentRootFileProvider).ConfigureAwait(false);
                 await DisposeAsync(_defaultProvider).ConfigureAwait(false);
             }
 
             // Dispose the service provider
             await DisposeAsync(Services).ConfigureAwait(false);
 
-            static async ValueTask DisposeAsync(object o)
+            static ValueTask DisposeAsync(object o)
             {
                 switch (o)
                 {
                     case IAsyncDisposable asyncDisposable:
-                        await asyncDisposable.DisposeAsync().ConfigureAwait(false);
-                        break;
+                        return asyncDisposable.DisposeAsync();
```

Comment:
There's no semantic change here, `DisposeAsync` callers continue to `await` with `ConfigureAwait(false)`.

Reply:
There is an unnecessary `await` + `ConfigureAwait(false)` for the `IAsyncDisposable` case but I think that may have been to prevent awaiting a default task for the `IDisposable` case?

Reply:
`default` for `ValueTask` is the same as `Task.CompletedTask` for `Task`